drivers/mpsl/clock_control/clock_control_mpsl.c

```diff
@@ -18,6 +18,14 @@
 #include <hal/nrf_power.h>
 #endif
 
+/* MPSL clock control structure */
+struct mpsl_clock_control_data {
+	sys_slist_t async_on_list;
```

Comment:
i don't see any reference counting. If there are multiple users of HF clock this will collapse and there at least 2: USB, NFC. It's even recommended for UART.

Driver should track number of requests and releases.

Reply:
The case of multiple users will be handled by 'mpsl_clock_hfclk_release' internally. The HF clock will be kept open until it is released by all users.

Reply:
Per discussion, before Zephyr 2.2 release, a reference counting is still needed in the driver. Therefore, I added a HFCLK-clock start counter  for short-term.